modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/ScaledFloatFieldMapper.java

```diff
@@ -446,17 +456,16 @@ protected void mergeOptions(FieldMapper other, List<String> conflicts) {
     protected void doXContentBody(XContentBuilder builder, boolean includeDefaults, Params params) throws IOException {
         super.doXContentBody(builder, includeDefaults, params);
 
-        builder.field("scaling_factor", fieldType().getScalingFactor());
+        builder.field("scaling_factor", scalingFactor);
 
         if (includeDefaults || ignoreMalformed.explicit()) {
             builder.field("ignore_malformed", ignoreMalformed.value());
         }
         if (includeDefaults || coerce.explicit()) {
             builder.field("coerce", coerce.value());
         }
-
-        if (includeDefaults || fieldType().nullValue() != null) {
-            builder.field("null_value", fieldType().nullValue());
+        if (nullValue != null) {
```

Comment:
is it ok that the includeDefaults condition is gone here?

Reply:
This is a slightly tricky one, in that `null` is not accepted as a value for `null_value` in mappings; so if we emit `null` here, then you can't use the output to set up mappings elsewhere.  FieldMapperTestBase in this PR adds a check that for each modified parameter, serialization/deserialization produces the same mapper configuration, both with and without including defaults, and this test will fail if `null` is emitted at this point.  It is a change, but I think it's a correct one?  I can open a separate issue regarding null values here, and mark it as `team-discuss` if we want to have a wider discussion about it.

Reply:
thanks for explaining, it sounds to me like no discussion is needed and this is a required change.